### PR TITLE
TEST/JENKINS: Avoid testing rdmacm on ib device with node_guid==0x0

### LINKED ISSUE #2534: UCX + RDMACM failure
```
#0  0x0000ffffb62b0578 in waitpid () from /lib64/libpthread.so.0
#1  0x0000ffffb678bc58 in ucs_debugger_attach () at ../../../src/ucs/debug/debug.c:703
#2  0x0000ffffb678d980 in ucs_error_freeze (message=0xffffb5e43000 "Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))", 
    error_type=0xffffb6821718 "address not mapped to object") at ../../../src/ucs/debug/debug.c:822
#3  ucs_handle_error (error_type=error_type@entry=0xffffb6821718 "address not mapped to object", message=message@entry=0xffffb6821408 "Caught signal %d (%s: %s%s)")
    at ../../../src/ucs/debug/debug.c:997
#4  0x0000ffffb678da64 in ucs_debug_handle_error_signal (signo=signo@entry=11, cause=0xffffb6821718 "address not mapped to object", 
    fmt=fmt@entry=0xffffb6821808 " at address %p") at ../../../src/ucs/debug/debug.c:934
#5  0x0000ffffb678dbc4 in ucs_error_signal_handler (signo=11, info=0xffffb5e437a0, context=<optimized out>) at ../../../src/ucs/debug/debug.c:957
#6  <signal handler called>
#7  0x0000ffffb64e6af8 in ucma_process_addr_resolved (evt=0x1c600070) at src/cma.c:2002
#8  rdma_get_cm_event (channel=0x1c6376a0, event=0xffffd6e175a0) at src/cma.c:2225
#9  0x0000ffffb64e6ca0 in rdma_get_cm_event (channel=<optimized out>, event=event@entry=0xffffd6e175a0) at src/cma.c:2303
#10 0x0000ffffb66ea214 in uct_rdmacm_get_event_type (event_ch=<optimized out>) at ../../../src/uct/ib/rdmacm/rdmacm_md.c:56
#11 uct_rdmacm_is_addr_route_resolved (timeout_ms=500, addr=<optimized out>, cm_id=<optimized out>) at ../../../src/uct/ib/rdmacm/rdmacm_md.c:84
#12 uct_rdmacm_is_sockaddr_accessible (md=0x1c600050, sockaddr=0xffffd6e176d0, mode=UCT_SOCKADDR_ACC_LOCAL) at ../../../src/uct/ib/rdmacm/rdmacm_md.c:146
#13 0x0000000000544fe0 in uct_test::set_sockaddr_resources (md=0x1c600050, md_name=0x1c5e72f0 "rdmacm", local_cpus=..., all_resources=...)
    at ../../../test/gtest/uct/uct_test.cc:131
#14 0x0000000000546d78 in uct_test::enum_resources (tl_name=..., loopback=loopback@entry=false) at ../../../test/gtest/uct/uct_test.cc:185
#15 0x000000000048c268 in gtest_rc_mlx5uct_amo_add_xor_test_EvalGenerator_ () at ../../../test/gtest/uct/test_amo_add_xor.cc:76
#16 0x0000000000490c74 in testing::internal::ParameterizedTestCaseInfo<uct_amo_add_xor_test>::RegisterTests (this=0xffffd6e17e50) at ../../../test/gtest/common/gtest.h:10495
#17 0x0000000000461f00 in RegisterTests (this=0x1c5b2f20) at /home/pavsha01/ucx/contrib/../test/gtest/common/gtest.h:10602
#18 RegisterParameterizedTests (this=0x1c5b2e30) at /home/pavsha01/ucx/contrib/../test/gtest/common/gtest-all.cc:3776
#19 testing::internal::UnitTestImpl::PostFlagParsingInit (this=0x1c5b2e30) at /home/pavsha01/ucx/contrib/../test/gtest/common/gtest-all.cc:5612
#20 0x00000000004620dc in PostFlagParsingInit (this=<optimized out>) at /var/nfsshare/sw/gcc-7.3/include/c++/7.3.0/bits/stl_construct.h:107
#21 InitGoogleTestImpl<char> (argv=0xffffd6e182a8, argc=0xffffd6e17fbc) at /home/pavsha01/ucx/contrib/../test/gtest/common/gtest-all.cc:6478
#22 testing::InitGoogleTest (argc=0xffffd6e17fbc, argc@entry=0xffffd6e17fdc, argv=argv@entry=0xffffd6e182a8)
    at /home/pavsha01/ucx/contrib/../test/gtest/common/gtest-all.cc:6493
#23 0x0000000000414bc4 in main (argc=<optimized out>, argv=0xffffd6e182a8) at ../../../test/gtest/common/main.cc:54
```
It fails here
```
#7  0x0000ffffb64e6af8 in ucma_process_addr_resolved (evt=0x1c600070) at src/cma.c:2002
```
Line
```
1997    static void ucma_process_addr_resolved(struct cma_event *evt)                                                                                                      |
   |1998    {                                                                                                                                                                  |
   |1999            if (af_ib_support) {                                                                                                                                       |
   |2000                    evt->event.status = ucma_query_addr(&evt->id_priv->id);                                                                                            |
   |2001                    if (!evt->event.status &&                                                                                                                          |
  >|2002                        evt->id_priv->id.verbs->device->transport_type == IBV_TRANSPORT_IB)                                                                            |
   |2003                            evt->event.status = ucma_query_gid(&evt->id_priv->id);                                                                                     |
   |2004            } else {                                                                                                                                                   |
   |2005                    evt->event.status = ucma_query_route(&evt->id_priv->id);                                                                                           |
   |2006            }                                                                                                                                                          |
   |2007                                                                                                                                                                       |
   |2008            if (evt->event.status)                                                                                                                                     |
   |2009                    evt->event.event = RDMA_CM_EVENT_ADDR_ERROR;                                                                                                       |
   |2010    } 
```

The bug:
```
#7  0x0000ffffb64e6af8 in ucma_process_addr_resolved (evt=0x1c600070) at src/cma.c:2002
(gdb) p   evt->id_priv->id.verbs
null
```

so verbs is null and obviously  it segfaults .

```
p evt->id_priv->id
$2 = {verbs = 0x0, channel = 0x1c6376a0, context = 0x0, qp = 0x0, route = {addr = {{src_addr = {sa_family = 2, sa_data = "\267M-MM-,\034a\370\000\000\000\000\000\000\000"}, 
        src_sin = {sin_family = 2, sin_port = 52663, sin_addr = {s_addr = 4167113900}, sin_zero = "\000\000\000\000\000\000\000"}, src_sin6 = {sin6_family = 2, 
          sin6_port = 52663, sin6_flowinfo = 4167113900, sin6_addr = {__in6_u = {__u6_addr8 = '\000' <repeats 15 times>, __u6_addr16 = {0, 0, 0, 0, 0, 0, 0, 0}, 
              __u6_addr32 = {0, 0, 0, 0}}}, sin6_scope_id = 0}, src_storage = {ss_family = 2, __ss_padding = "\267M-MM-,\034a\370", '\000' <repeats 111 times>, __ss_align = 0}}
, {
        dst_addr = {sa_family = 2, sa_data = "\000\000\254\034a\370\000\000\000\000\000\000\000"}, dst_sin = {sin_family = 2, sin_port = 0, sin_addr = {s_addr = 4167113900}, 
          sin_zero = "\000\000\000\000\000\000\000"}, dst_sin6 = {sin6_family = 2, sin6_port = 0, sin6_flowinfo = 4167113900, sin6_addr = {__in6_u = {
              __u6_addr8 = '\000' <repeats 15 times>, __u6_addr16 = {0, 0, 0, 0, 0, 0, 0, 0}, __u6_addr32 = {0, 0, 0, 0}}}, sin6_scope_id = 0}, dst_storage = {ss_family = 2, 
          __ss_padding = "\000\000\254\034a\370", '\000' <repeats 111 times>, __ss_align = 0}}, addr = {ibaddr = {sgid = {
            raw = "\000\000\000\000\000\000\000\000\000\000\377\377\254\034", <incomplete sequence \370>, global = {subnet_prefix = 0, interface_id = 17897617923501916160}}, 
          dgid = {raw = '\000' <repeats 15 times>, global = {subnet_prefix = 0, interface_id = 0}}, pkey = 0}}}, path_rec = 0x0, num_paths = 0}, ps = RDMA_PS_UDP, 
  port_num = 0 '\000', event = 0x0, send_cq_channel = 0x0, send_cq = 0x0, recv_cq_channel = 0x0, recv_cq = 0x0, srq = 0x0, pd = 0x0, qp_type = IBV_QPT_UD}

p evt->event        
$4 = {id = 0x1c600aa0, listen_id = 0x0, event = RDMA_CM_EVENT_ADDR_RESOLVED, status = 0, param = {conn = {private_data = 0x0, private_data_len = 0 '\000', 
      responder_resources = 0 '\000', initiator_depth = 0 '\000', flow_control = 0 '\000', retry_count = 0 '\000', rnr_retry_count = 0 '\000', srq = 0 '\000', qp_num = 0}, 
    ud = {private_data = 0x0, private_data_len = 0 '\000', ah_attr = {grh = {dgid = {raw = '\000' <repeats 15 times>, global = {subnet_prefix = 0, interface_id = 0}}, 
          flow_label = 0, sgid_index = 0 '\000', hop_limit = 0 '\000', traffic_class = 0 '\000'}, dlid = 0, sl = 0 '\000', src_path_bits = 0 '\000', static_rate = 0 '\000', 
        is_global = 0 '\000', port_num = 0 '\000'}, qp_num = 0, qkey = 0}}}
```
at
```
120     void uct_test::set_sockaddr_resources(uct_md_h md, char *md_name, cpu_set_t local_cpus,                                                                            |
   |121                                           std::vector<resource>& all_resources) {  
```
I see that it fails on "lo" / loopback inteface:
```
 p *ifaddr
$14 = {ifa_next = 0x1c6002f8, ifa_name = 0x1c6002e4 "lo", ifa_flags = 65609, ifa_addr = 0x1c600278, ifa_netmask = 0x0, ifa_ifu = {ifu_broadaddr = 0x1c6002c0, 
    ifu_dstaddr = 0x1c6002c0}, ifa_data = 0x1c6008b8}
```

So it is something between rdmacm and ucx. Apparently it does not like "lo" interface.

RHEL7.5, OFED-internal-4.3-1.0.1, UCX master

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #2534

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6966/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6967/ for details (Mellanox internal link).

File: test/gtest/common/test_helpers.cc [hunk 1]
```diff
@@ -76,17 +76,39 @@ bool is_inet_addr(const struct sockaddr* ifa_addr) {
...
     if (dir == NULL) {
         return false;
-    } else {
-        closedir(dir);
-        return true;
     }
+
+    /* read IB device name */
+    for (;;) {
+        struct dirent *entry = readdir(dir);
```
Comment (alinask) [inline test/gtest/common/test_helpers.cc]: why declaring 'entry' here and not in the beginning of the function?

File: test/gtest/common/test_helpers.cc [hunk 2]
```diff
@@ -76,17 +76,39 @@ bool is_inet_addr(const struct sockaddr* ifa_addr) {
 
 bool is_ib_netdev(const char *ifa_name) {
```
Comment (alinask) [inline test/gtest/common/test_helpers.cc]: can you please update the documentation of this function in the .h file now that it also checks guid?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6971/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 3]
```diff
@@ -76,7 +76,7 @@ class test_ucp_sockaddr : public ucp_test {
         for (struct ifaddrs *ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
             if (ucs_netif_is_active(ifa->ifa_name) &&
                 ucs::is_inet_addr(ifa->ifa_addr)   &&
-                ucs::is_ib_netdev(ifa->ifa_name))
+                ucs::is_rdmacm_netdev(ifa->ifa_name))
```
Comment (alinask) [inline test/gtest/ucp/test_ucp_sockaddr.cc]: should we be mentioning an rdmacm limitation in a ucp test..?
i mean, when we add another sockaddr transport, this check will exclude the new transport

Comment (yosefe) [reply test/gtest/ucp/test_ucp_sockaddr.cc]: i don't think it's worse than mentioning 'ib', at least it's more accurate
if we add another sockaddr transport will have to update this check

Comment (alinask) [reply test/gtest/ucp/test_ucp_sockaddr.cc]: IMHO we should be generic, like we do in the sockaddr_accessibility test in test_md.c, but ok, your call.

**Pull request merged**